sympy/sets/ordinals.py

```diff
@@ -260,8 +275,14 @@ class OrdinalOmega(Ordinal):
     >>> omega + omega
     w*2
     """
-    def __new__(cls):
-        return Ordinal.__new__(cls, OmegaPower(1, 1))
 
-ord0 = OrdinalZero()
+    # Override terms property of Ordinal
+    terms = (OmegaPower(1, 1),)
```

Comment:
The OrdinaOmage class does not satisfy the `x == x.func(*x.args)` invariant. I have changed it here so that `OrdinalOmega` takes no args but a terms property is used to make this consistent with its super class. This mechanism is similar to the one used in Reals.